### DISCUSSION TITLE
Dose calculation when use range shifter

### DISCUSSION THREAD

**Discussion by zeng99-jiu** [e0404/matRad#593] [Q&A] [answered]
I wanna ask if the "matRad_calcParticleDose.m" file has considered that the range shifters change the beam current and energy spectrum   before the particle enter the body.

**Comment by wahln** [comment, marked as answer]
For the pencil-beam algorithm, not really. The range shifter internally changes the WEPL and adds a correction to the beam broadening before calculating the dose distribution of an individual spot using the tabulated depth dose. Beam _current_ (meaning in [A] or protons/s) is not explicitly modeled, only intensity (i.e. total number of protons per beamlet). Thus, the range & deposited dose will change as it would have through water of equivalent depth (the detailed spectral energy deposition specific to the range shifter material is not modeled). The lateral broadening will assume a more realistic material of the range shifter (I think the default is PMMA).  
For the Monte Carlo interfaces on the dev_varRBErobOpt branch, the range shifter is placed and simulated as a physical object, but I presume here some hacking is required to get the material exactly right.

**Reply by zeng99-jiu** [reply]
I got it! Thanks for your reply！
